### PR TITLE
relax throws clause in overridden methods that do not throw

### PR DESCRIPTION
**Pull request by elsazac**
Refs: https://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html#jls-8.4.8.3

### CODE REVIEW

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/browser/DefaultWorkbenchBrowserSupport.java [hunk 1]
```diff
@@ -53,7 +53,7 @@ IWebBrowser findBrowser(String id) {
 	}
 
 	protected IWebBrowser doCreateBrowser(int style, String browserId, String name, String tooltip)
```
Comment (laeubi) [inline bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/browser/DefaultWorkbenchBrowserSupport.java:54]: Please check the java warning

**Review by laeubi — DISMISSED** [review]
It seems you need to adjust the javadoc as well then

**Comment by elsazac** [comment]
> It seems you need to adjust the javadoc as well then

@laeubi looks like the javadoc of the overridden methods are dynamically adjusted as per the declaration.

before:
image (1)

after:
image (2)

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/browser/DefaultWorkbenchBrowserSupport.java (hunk 1 above)
Comment (elsazac) [reply bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/browser/DefaultWorkbenchBrowserSupport.java:54]: The java warnings are preexisting.

before:
image (3)

image (5)


after:
image (6)

**Comment by laeubi** [comment]
Please check the build that shows new warnings, this maybe because you removed the exception from one place then another one becomes no longer throwing the exception as for example here:

https://ci.eclipse.org/platform/job/eclipse.platform.ui/job/PR-1467/1/java/new/source.ddf8b0c4-8449-4f37-a47f-0accdbbf3117/#62

**Comment by elsazac** [comment]
> Please check the build that shows new warnings, this maybe because you removed the exception from one place then another one becomes no longer throwing the exception as for example here:
> 
> https://ci.eclipse.org/platform/job/eclipse.platform.ui/job/PR-1467/1/java/new/source.ddf8b0c4-8449-4f37-a47f-0accdbbf3117/#62

@laeubi Thanks for the clarification. Now I see where it is coming from. please have a look.

**Comment by laeubi** [comment]
Build still shows one new compiler warning even though Jenkins seem to have problems parsing them:

https://ci.eclipse.org/platform/job/eclipse.platform.ui/job/PR-1467/3/java/new/type.1251191584/

**Comment by elsazac** [comment]
> Please check the build that shows new warnings, this maybe because you removed the exception from one place then another one becomes no longer throwing the exception as for example here:
> 
> https://ci.eclipse.org/platform/job/eclipse.platform.ui/job/PR-1467/1/java/new/source.ddf8b0c4-8449-4f37-a47f-0accdbbf3117/#62

@laeubi we did a detailed investigation, and could not find any new warnings.
 - no of preexisting warnings : 2374
 - no of throws clause removed : 3
 - no of additional warnings introduced : 2
 - and hence no of additional throws clause removed : 2
 - no of current (with this commit) warnings : 2371
 
 I undone everything, and introduce the changes one by one and confirm that there is no additional warnings introduced.The yellow strip that Jenkins shows should contain three preexisting warnings (on unused parameters), probably it was hidden behind the throws clause warning and got highlighted after this commit - that's my theory.

**Comment by laeubi** [comment]
> The yellow strip that Jenkins shows should contain three preexisting warnings (on unused parameters), probably it was hidden behind the throws clause warning and got highlighted after this commit

Its possible that this is now counted as new, can you just fix the warnings there?

**Comment by elsazac** [comment]
> > The yellow strip that Jenkins shows should contain three preexisting warnings (on unused parameters), probably it was hidden behind the throws clause warning and got highlighted after this commit
> 
> Its possible that this is now counted as new, can you just fix the warnings there?

``` java
protected IWebBrowser doCreateBrowser(int style, String browserId, String name, String tooltip)
	{
		return new DefaultWebBrowser(this, browserId);
	}
```
this is the code. The warning complaints about unused params. I refrained from removing the params because it's a protected method (overridden methods may choose to use these params). what do you suggest?

**Comment by merks** [comment]
Annotations can be used:

image

**Review by mickaelistria — APPROVED** [review]
Code is better, no new warning introduced, let's merge.
Other issues can be considered seperately (more easier tracking and better productivity)

**Comment by laeubi** [comment]
`DefaultWorkbenchBrowserSupport` is an internal class so we can change it anytime and anyway...

**Pull request merged**